### ISSUE TITLE
New versions are signed with different key on F-Droid

### ISSUE THREAD

**Issue by bytemybits** [wallabag/android-app#614]
Whenever I try to update the app through F-Droid I get the message stating that the new version is signed with a different key, therefore I need to re-install the app. This happened from the last 1.x to 2.x and I thought maybe because it major was a version (strange I know), but now is also happening for a minor 2.0.0 -> 2.0.1.
Is there any special reason for this?

**Comment by di72nn** [comment]
Are you using CopperheadOS (if not, what is your Android version and which flavor it is (AOSP, OEM, or some third-party build))? There was a report about something like that from a CopperheadOS user. However, I'm not aware if that was solved.

We did not change anything regarding F-Droid signing the wallabag apks ([current metadata](https://gitlab.com/fdroid/fdroiddata/blob/master/metadata/fr.gaulupeau.apps.InThePoche.txt), [history](https://gitlab.com/fdroid/fdroiddata/commits/master/metadata/fr.gaulupeau.apps.InThePoche.txt)).
I updated wallabag on my phone (using F-Droid) from 1.12.2 to 2.0.0 and from 2.0.0 to 2.0.1 without any problems (I did it when the corresponding versions were released).

Also I've just downloaded the latest three versions from F-Droid and they seem to be signed with the same certificate.

So it is likely to be F-Droid app issue.

Can you download the latest wallabag apk from F-Droid (via web-browser) and install it on your phone manually? If it really has a different signature, your phone should not allow it.

**Comment by bytemybits** [comment]
I'm using a CyanogenMod build (14.1-20161119-NIGHTLY-bacon). I've tried downloading the apk manually and it didn't install because the signature is in fact different. That's the same reason I can't update the app through F-Droid. This is strange to me because no other app I use from F-Droid has given me this problem.

**Comment by di72nn** [comment]
Can you extract currently installed APK from your phone and upload it?

**Comment by bytemybits** [comment]
Here it is:
[wallabag_fr.gaulupeau.apps.InThePoche.apk.zip](https://github.com/wallabag/android-app/files/1158363/wallabag_fr.gaulupeau.apps.InThePoche.apk.zip)

**Comment by tcitworld** [comment]
We now use https://source.android.com/security/apksigning/v2 while signing Google Play / Github apks, but it shouldn't be related anyway.

**Comment by di72nn** [comment]
The file matches [the 2.0.0 release](https://github.com/wallabag/android-app/releases/tag/2.0.0) available from our release page and from Google Play. Are you sure you installed it from F-Droid?

**Comment by bytemybits** [comment]
I tried downloading the apk from PlayStore and it did install. I'm almost sure that I downloaded it from F-Droid as I always do whenever possible since I don't even have the PlayStore installed.
Anyway, I uninstalled and installed 1.12.2 from F-Droid and upgraded twice until 2.0.1. It worked flawlessly!
I'm sorry for bothering with something that apparently wasn't an issue. Thank you for the help. 🙂
